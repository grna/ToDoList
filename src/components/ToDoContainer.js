import React from "react";
import PropTypes from "prop-types";
import ToDoList from "./ToDoList";
import { connect } from "react-redux";
import "../styles/todo.css";
import {
  addNewTask,
  updateTask,
  deleteTask,
} from "../redux/actions/tasksActions";

const ToDoContainer = ({ tasks, addNewTask, updateTask, deleteTask }) => {
  const notStartedTasks = tasks.filter((todo) => {
    return todo.status === "not started";
  });

  const inProgressTasks = tasks.filter((todo) => {
    return todo.status === "in progress";
  });

  const doneTasks = tasks.filter((todo) => {
    return todo.status === "done";
  });

  return (
    <div className="container">
      <div className="not-started column">
        <div className="todo-header">
          <h3>Not Started</h3>
          <button className="todo-add-btn" onClick={() => addNewTask()}>
            +
          </button>
        </div>
        {notStartedTasks.length > 0 ? (
          <ToDoList
            className="not-started"
            toDoList={notStartedTasks}
            updateTask={updateTask}
            deleteTask={deleteTask}></ToDoList>
        ) : (
          <div></div>
        )}
      </div>
      <div className="in-progress column">
        <div className="todo-header">
          <h3>In Progress</h3>
        </div>
        {inProgressTasks.length > 0 ? (
          <ToDoList
            className="in-progress"
            toDoList={inProgressTasks}
            updateTask={updateTask}
            deleteTask={deleteTask}></ToDoList>
        ) : (
          <div></div>
        )}
      </div>
      <div className="done column">
        <div className="todo-header">
          <h3>Done</h3>
        </div>
        {doneTasks.length > 0 ? (
          <ToDoList
            className="done"
            toDoList={doneTasks}
            updateTask={updateTask}
            deleteTask={deleteTask}></ToDoList>
        ) : (
          <div></div>
        )}
      </div>
    </div>
  );
};

ToDoContainer.propTypes = {
  tasks: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      status: PropTypes.string.isRequired,
    })
  ),
  addNewTask: PropTypes.func.isRequired,
  updateTask: PropTypes.func.isRequired,
  deleteTask: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => ({
  tasks: state.fromTasks.tasks,
});

export default connect(mapStateToProps, { addNewTask, updateTask, deleteTask })(
  ToDoContainer
);
