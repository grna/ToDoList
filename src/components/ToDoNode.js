import React, { useState } from "react";
import PropTypes from "prop-types";

function ToDoNode({ todo, updateTask }) {
  const [title, setTitle] = useState(todo.title);
  const [description, setDescription] = useState(todo.description);

  const onTaskUpdate = () => {
    const newTask = {
      _id: todo._id,
      title: title,
      description: description,
      status: todo.status,
    };

    updateTask(newTask);
  };

  const onTaskDelete = () => {};

  return (
    <div className="todo-node">
      <label>
        <strong>Title:</strong>
      </label>
      <input
        onChange={(e) => setTitle(e.target.value)}
        type="text"
        value={title}
      />
      <label>
        <strong>Description:</strong>
      </label>

      <input
        onChange={(e) => setDescription(e.target.value)}
        type="text"
        value={description}
      />
      <label>
        <strong>Status: </strong>
        {todo.status}
      </label>
      <button onClick={() => onTaskUpdate()}>Update</button>
      <button onClick={() => onTaskDelete()}>Delete</button>
    </div>
  );
}

ToDoNode.propTypes = {
  todo: PropTypes.shape({
    _id: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
    status: PropTypes.string,
  }),
  updateTask: PropTypes.func,
};

export default ToDoNode;
