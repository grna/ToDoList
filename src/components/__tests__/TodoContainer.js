import { mapStateToProps } from "../ToDoContainer";
import renderer from "react-test-renderer";
import React from "react";
import ToDoNode from "../ToDoNode";

describe("The ToDo Container", () => {
  describe("mapStateToProps", () => {
    it("Should map state to props", () => {
      const sampleTask = {
        _id: "task1",
        title: "Sample task",
        description: "Sample task description",
        status: "in progress",
      };
      const state = {
        fromTasks: {
          tasks: [sampleTask],
        },
      };
      const componentState = mapStateToProps(state);
      expect(componentState.tasks[0]).toEqual(sampleTask);
    });
  });
});

describe("ToDoNode component", () => {
  it("Should not regress", () => {
    const todo = {
      _id: "task1",
      title: "Sample task",
      description: "Sample task description",
      status: "in progress",
    };
    const updateTask = () => {};
    const deleteTask = () => {};
    const tree = renderer.create(
      <ToDoNode todo={todo} updateTask={updateTask} deleteTask={deleteTask} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
