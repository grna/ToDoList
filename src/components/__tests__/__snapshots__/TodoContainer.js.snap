// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoNode component Should not regress 1`] = `
<div
  className="todo-node"
>
  <label>
    <strong>
      Title:
    </strong>
  </label>
  <input
    onChange={[Function]}
    type="text"
    value="Sample task"
  />
  <label>
    <strong>
      Description:
    </strong>
  </label>
  <textarea
    defaultValue="Sample task description"
    onChange={[Function]}
  />
  <label>
    <strong>
      Status: 
    </strong>
  </label>
  <select
    defaultValue="in progress"
    onChange={[Function]}
  >
    <option
      className="task-status"
      value="not started"
    >
      Not Started
    </option>
    <option
      className="task-status"
      value="in progress"
    >
      In Progress
    </option>
    <option
      className="task-status"
      value="done"
    >
      Done!
    </option>
  </select>
  <div
    className="todo-buttons"
  >
    <button
      onClick={[Function]}
    >
      Save
    </button>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
